github/PullRequest.py

```diff
@@ -775,6 +775,19 @@ def merge(self, commit_message=github.GithubObject.NotSet, commit_title=github.G
             self.url + "/merge",
             input=post_parameters
         )
+
+
+        if delete_branch:
+
+            # Check for other pull requests referencing the same ref.
+            # If found, abort.
+            remaining_pulls = self.head.repo.get_pulls(head=self.head.ref)
+            if remaining_pulls:
```

Comment:
What if I want to force delete a branch? Then `delete_branch=True` will not work, if some other PR is still referencing it right? If this the intended behavior, I am not so sure.

Reply:
Not sure.  I don't think merging one PR should close other PRs without explicit confirmation. 
 Perhaps deleting a branch should be done via a separate call?  So I take the code I added and put it in a `delete_branch(force=False)` method? 

So typical usage would be
```
pr = repo.get_pull(...)
pr.merge(...)
pr.delete_branch(force=True)
```

Reply:
Actually that would make a lot of sense, because then the api for restoring a branch is similar.